test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_image_arguments.cpp

```diff
@@ -0,0 +1,432 @@
+    virtual cl_int SetUp(int elements) override
+    {
+        BasicMutableCommandBufferTest::SetUp(elements);
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        cl_bool image_support;
+
+        cl_int error =
+            clGetDeviceInfo(device, CL_DEVICE_IMAGE_SUPPORT,
+                            sizeof(image_support), &image_support, nullptr);
+        test_error(error, "clGetDeviceInfo for CL_DEVICE_IMAGE_SUPPORT failed");
+
+        cl_mutable_dispatch_fields_khr mutable_capabilities;
+
+        bool mutable_support =
+            !clGetDeviceInfo(
+                device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+                sizeof(mutable_capabilities), &mutable_capabilities, nullptr)
+            && mutable_capabilities & CL_MUTABLE_DISPATCH_LOCAL_SIZE_KHR;
+
+        return (!mutable_support && !image_support)
+            || BasicMutableCommandBufferTest::Skip();
+    }
+
+    cl_int Run() override
+    {
+
+        const char *sample_const_arg_kernel =
+            R"(__kernel void sample_test( read_only image2d_t source, sampler_t
+            sampler, write_only image2d_t dest)
+            {
+               int x = get_global_id(0);
+               int y = get_global_id(1);
+
+               int4 color = read_imagei( source, sampler, (int2) (x, y) );
+
+               write_imagei( dest, (int2) (x, y), color );
+            })";
+
+        cl_int error;
+        clProgramWrapper program;
+        clKernelWrapper kernel;
+
+        cl_image_desc image_desc;
+        memset(&image_desc, 0x0, sizeof(cl_image_desc));
+        image_desc.image_type = CL_MEM_OBJECT_IMAGE2D;
+        image_desc.image_width = 4;
+        image_desc.image_height = 4;
+        image_desc.image_row_pitch = 0;
+        image_desc.num_mip_levels = 0;
+
+        size_t data_size =
+            image_desc.image_width * image_desc.image_height * sizeof(cl_int);
+
+        const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+
+        image_descriptor imageInfo = { 0 };
+        imageInfo.type = CL_MEM_OBJECT_IMAGE2D;
+        imageInfo.width = 4;
+        imageInfo.height = 4;
+        imageInfo.format = &formats;
+
+        BufferOwningPtr<char> imageValues;
+
+        MTdataHolder d(gRandomSeed);
+        generate_random_image_data(&imageInfo, imageValues, d);
+
+        char *host_ptr = (char *)imageValues;
+        std::vector<char> outputData(data_size);
+
+        clMemWrapper src_image =
+            create_image_2d(context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR,
+                            &formats, image_desc.image_width,
+                            image_desc.image_height, 0, host_ptr, &error);
+        test_error(error, "create_image_2d failed");
+
+        clMemWrapper dst_image =
+            create_image_2d(context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR,
+                            &formats, image_desc.image_width,
+                            image_desc.image_height, 0, host_ptr, &error);
```

Comment:
Same comment about using different data to initalize `dst_image` than is in `src_image`